### DISCUSSION TITLE
Support for local media keys for play/pause/skip etc

### DISCUSSION THREAD

**Discussion by nullr0ute** [librespot-org/librespot#756] [Feature Requests]
I couldn't see anything obvious in docs etc. Is there a way to be able to use media keys to control music. This could be the KEY_PLAYPAUSE/KEY_NEXTSONG style Linux media input events. That way a user could point librespot at an input device, such as a CEC remote control, or a GPIO keypad with those keys such as comes with some of the Raspberry Pi audio HATs to get local control.

**Comment by ashthespy** [comment]
https://github.com/Spotifyd/spotifyd supports Dbus MPRIS -- maybe that is a better option?

**Reply by nullr0ute** [reply]
Doesn't really work in a headless environment. MPRIS is expecting a UX such as desktop environment listening to key events which it then decodes the media keys and sends them over dbus to the media player. For that to work in a headless audio style usecase you'd have to have some form of input event listener which would then have to pass over dbus. It seems like a lot of layers of indirection when there's no desktop UX that is listening for key events anyway.

**Reply by ashthespy** [reply]
There isn't a way to send direct control to `librespot`(the binary)  it's a pure connect implementation, so playback controls need to come via Connect, i.e either from another Spotify Client, or using the WebAPI. 


You could use `librespot` the library, and build something more minimal that Spotfyd

**Reply by kingosticks** [reply]
This isn't going to help you but a CLI program like `mpc` built on librespot for controlling Connect instances would be awesome. IMO the benefit of building something like this on librespot would be the ability to use simple user/pass login rather than messing around registering an app and doing an OAuth dance as required with something built around Spotify's Web API. [SpotControl](https://github.com/badfortrains/spotcontrol/tree/master/examples/micro-controller) has something like this but I don't know how advanced it is. Then it's a case of wiring up such a CLI program with the likes of lirc, [cecdaemon](https://github.com/simons-public/cecdaemon) or something like [inputexec](https://github.com/rbarrois/inputexec).

And for what it's worth, MPRIS can be used without a full desktop session. Things like `dbus-launch` and `dbus-run-session` are available for this ([example](https://gist.github.com/ritiek/0de8d57a1a77c3ea47ede0a71ea6afcc)). But you still have to handle catching the keypresses yourself (or use inputexec).
